### PR TITLE
fix: False positive NoLogsDelivered LogPipeline status reason

### PR DESCRIPTION
**Pull request by skhalash**
## Description

Changes proposed in this pull request (what was done and why):

- Before the `NoLogsDelievered` status reason was calculated based on 2 alerts: `LogAgentReceiverReadLogs` and `LogAgentExporterSentLogs`. This caused some flakiness when both alert statuses were not updated simultaneously. In this PR, a new alert was introduced that is logically the same, but boolean logic is done in PromQL instead of the Go code. Moreover, the `for` alert attribute was set to 1 minute to prevent flakiness.

Changes refer to particular issues, PRs or documents:

- https://github.com/kyma-project/telemetry-manager/issues/1358

## Traceability
- [x] The PR is linked to a GitHub issue.
- [ ] The follow-up issues (if any) are linked in the `Related Issues` section.
- [ ] If the change is user-facing, the documentation has been adjusted.
- [x] The feature is unit-tested.
- [x] The feature is e2e-tested.

### CODE REVIEW

File: internal/selfmonitor/config/fluent_bit_rule_builder.go [hunk 1]
```diff
@@ -74,6 +71,24 @@ func (rb fluentBitRuleBuilder) bufferFullRule() Rule {
...
 
+func (rb fluentBitRuleBuilder) noLogsDeliveredRule() Rule {
+	receiverReadExpr := rate(metricFluentBitInputBytesTotal, selectService(fluentBitMetricsServiceName)).
+		sumBy(labelPipelineName).
+		greaterThan(0).
+		build()
+
+	exporterNotSentExpr := rate(metricFluentBitOutputProcBytesTotal, selectService(fluentBitMetricsServiceName)).
+		sumBy(labelPipelineName).
+		equal(0).
```
Comment (TeodorSAP) [inline internal/selfmonitor/config/fluent_bit_rule_builder.go:82]: Maybe it's a stupid idea, but wouldn't it be safer to use "<="

File: internal/selfmonitor/prober/log_pipeline_prober_test.go [hunk 2]
```diff
@@ -61,6 +61,25 @@ func TestLogPipelineProber(t *testing.T) {
...
 			},
 		},
+		{
+			name:         "pending alert",
+			pipelineName: "cls",
+			alerts: promv1.AlertsResult{
+				Alerts: []promv1.Alert{
+					{
+						Labels: model.LabelSet{
+							"alertname": "LogAgentBufferFull",
```
Comment (TeodorSAP) [inline internal/selfmonitor/prober/log_pipeline_prober_test.go:71]: What is this test case testing? Are we using the pending state on the LogAgentBufferFull as well?

Comment (TeodorSAP) [reply internal/selfmonitor/prober/log_pipeline_prober_test.go:71]: Shouldn't this test target the "NoLogsDelivered" pending state?

File: internal/selfmonitor/config/fluent_bit_rule_builder.go (hunk 1 above)
Comment (skhalash) [reply internal/selfmonitor/config/fluent_bit_rule_builder.go:82]: Rate cannot be negative

File: internal/selfmonitor/prober/log_pipeline_prober_test.go (hunk 2 above)
Comment (skhalash) [reply internal/selfmonitor/prober/log_pipeline_prober_test.go:71]: No, this is the whole idea of this change. If an alert is pending, we should not map to the condition reason, only if it's firing.

Comment (TeodorSAP) [reply internal/selfmonitor/prober/log_pipeline_prober_test.go:71]: Ok, but here you are checking that if it's pending, `Healthy: true` right?

**Review by TeodorSAP — APPROVED** [review]

**Pull request merged**